rtgui/editcallbacks.h

```diff
@@ -0,0 +1,256 @@
+    @return true if the preview has to be redrawn, false otherwise */
+    virtual bool button3Pressed (const int modifierKey);
+
+    /** @brief Triggered when mouse button 3 is released (right button)
+    @return true if the preview has to be redrawn, false otherwise */
+    virtual bool button3Released ();
+
+    /** @brief Triggered when the user is moving while holding down mouse button 1
+    @param modifierKey Gtk's event modifier key (GDK_CONTROL_MASK | GDK_SHIFT_MASK | ...)
+    @return true if the preview has to be redrawn, false otherwise */
+    virtual bool drag1 (const int modifierKey);
+
+    /** @brief Triggered when the user is moving while holding down mouse button 2
+    @param modifierKey Gtk's event modifier key (GDK_CONTROL_MASK | GDK_SHIFT_MASK | ...)
+    @return true if the preview has to be redrawn, false otherwise */
+    virtual bool drag2 (const int modifierKey);
+
+    /** @brief Triggered when the user is moving while holding down mouse button 3
+    @param modifierKey Gtk's event modifier key (GDK_CONTROL_MASK | GDK_SHIFT_MASK | ...)
+    @return true if the preview has to be redrawn, false otherwise */
+    virtual bool drag3 (const int modifierKey);
+
+    /** @brief Triggered when the user is releasing mouse button 1 while in action==ES_ACTION_PICKING mode
+    No modifier key is provided, since having a different modifier key than on button press will set picked to false.
+    @param picked True if the cursor is still above the the same object than on button pressed and with the same modifier keys.
+                  If false, the user moved the cursor away or the modifier key is different, so the element is considered as NOT selected.
+    @return true if the preview has to be redrawn, false otherwise */
+    virtual bool pick1 (const bool picked);
+
+    /** @brief Triggered when the user is releasing mouse button 2 while in action==ES_ACTION_PICKING mode
+    @param picked True if the cursor is still above the the same object than on button pressed and with the same modifier keys.
+                  If false, the user moved the cursor away or the modifier key is different, so the element is considered as NOT selected.
+    @return true if the preview has to be redrawn, false otherwise */
+    virtual bool pick2 (const bool picked);
+
+    /** @brief Triggered when the user is releasing mouse button 3 while in action==ES_ACTION_PICKING mode
+    @param picked True if the cursor is still above the the same object than on button pressed and with the same modifier keys.
+                  If false, the user moved the cursor away or the modifier key is different, so the element is considered as NOT selected.
+    @return true if the preview has to be redrawn, false otherwise */
+    virtual bool pick3 (const bool picked);
+
+    /** @brief Get the geometry to be shown to the user */
+    const std::vector<Geometry*>& getVisibleGeometry ();

```

Comment:
Is the caller allowed to change the geometries? Does the calling change the `EditSubscriber` instance? If the answer is "no" twice consider this:

```c++
std::vector<const Geometry*> getVisibleGeometry() const;
```